pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -19,40 +19,42 @@ package sidecarmetricscollector
 import (
 	"os"
 	"path/filepath"
-	"reflect"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	commonv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 )
 
-var testJsonDataPath = filepath.Join("testdata", "JSON")
+var (
+	testJsonDataPath = filepath.Join("testdata", "JSON")
+	testTextDataPath = filepath.Join("testdata", "TEXT")
+)
 
 func TestCollectObservationLog(t *testing.T) {
-
-	if err := generateTestFiles(); err != nil {
+	if err := generateJSONTestFiles(); err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(filepath.Dir(testJsonDataPath))
+	if err := generateTEXTTestFiles(); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(filepath.Dir(testTextDataPath))
 
-	// TODO (tenzen-y): We should add tests for logs in TEXT format.
-	// Ref: https://github.com/kubeflow/katib/issues/1756
-	testCases := []struct {
-		description string
-		filePath    string
-		metrics     []string
-		filters     []string
-		fileFormat  commonv1beta1.FileFormat
-		err         bool
-		expected    *v1beta1.ObservationLog
+	testCases := map[string]struct {
```

Comment:
Thank you for refactoring existing tests!

Reply:
Haha, ust follow the `Developer Guide`.